p2p/libp2p/networksharding/kadSharder.go

```diff
@@ -15,6 +15,7 @@ const (
 // KadPeerShardResolver peer to shard mapping interface
 type KadPeerShardResolver interface {
 	ByID(p2p.PeerID) uint32 //ByID get the shard id of the given peer.ID
+	IsBalanced() bool       //IsBalanced checks if the sharding is balanced and connections initiation can be paused
```

Comment:
This can not be computed by the Mapper implementation. It simply does not have the required info to decide nor it should care how the lower level component manage its connections. I think this should be resolved by Sharded implementations. For example, on the bit masking kadSharder implementation, while iterating the sortingList and calling resolver.ByID, you could also have some counters and then decide if the the list is balanced or not.

Reply:
done, but I'll need the total number of shards

Reply:
That can be provided by the KadPeerShardResolver as it a high level info. Just add a function like:
```GO
NumberOfShards() uint32
```